client/ClientCommandManager.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * ClientCommandManager.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#pragma once
+
+VCMI_LIB_NAMESPACE_BEGIN
+class PlayerColor;
+VCMI_LIB_NAMESPACE_END
+class CIntObject;
+
+class ClientCommandManager //take mantis #2292 issue about account if thinking about handling cheats from command-line
+{
+	static bool currentCallFromIngameConsole;
+
+	static void giveTurn(const PlayerColor &color);
+	static void printInfoAboutInterfaceObject(const CIntObject *obj, int level);
+	static void printCommandMessage(const std::string &commandMessage, ELogLevel::ELogLevel messageType = ELogLevel::NOT_SET);
+	static void handleGoSolo();
+	static void handleControlAi(const std::string &colorName);
+
+public:
+	ClientCommandManager() = delete;
+	static void processCommand(const std::string &message, bool calledFromIngameConsole);
+};
```

Comment:
it doesn't manage something, it's event not a class but set of functions. So why not have set of functions wrapped by namespace?

Reply:
It does use "private" visibility to hide internal stuff from outside calls.
Perhaps remove 'static' and use it as proper class instead?
As for name - Controller or Executor perhaps?

Reply:
so you recommend creating new instance for each usage?

Reply:
In my opinion - yes, but on stack, without call to new

Reply:
Unsure about that yet, since this class is context independent and ability to freely create its object doesn't bring anything interesting to the table. The internal state isn't interesting for caller. Command can be executed at any arbitrary place in code so being static makes sense, from "domain driven design" point of view etc.

Reply:
Personally don't care either way, as long as it stays as "class" - it is weird to have a namespace with random functions that should not be called.
Nordsoft didn't liked this "static class" approach, so I suggested alternative.

Reply:
well, in theory it could be singleton class, but does it matter? For example in C# you have Math class with bunch of static methods, Unity engine adds Random with another static methods (C# standard libraries also has Random but it's non static for some reason and often wrapped in some singleton so you don't create new instances over and over for the sake of good RNG seed)

Reply:
Keep in mind that C#
1) Does not have concept of "namespace with functions", all functions must be part of a class, so Nordsoft proposition is something that only exists in C++
2) I assume C# uses singleton in these cases to avoid (relatively) costly object creation call. In C++, creation on stack is extremely cheap operation. Especially for a state-less class.
3) Singleton is often viewed as an anti-pattern. Our VLC would be a great example of such.

By the way, your class is *not* stateless due to `currentCallFromIngameConsole` member. So yeah, I also suggest to change this, but into a proper class, with allocation on stack.

Reply:
Done, though the currentCallFromIngameConsole is there for convenience and it is rewritten on every public function call - I just do not want to pass bool as argument everywhere within class

Reply:
Sure, my complain against `currentCallFromIngameConsole` was mostly about having it as static (which is now fixes)
While this is close to impossible with current design, if you somehow do end up with two instances of your class, you could end up with them accessing same variable since it was static.

Reply:
would be still technically impossible under normal conditions for one client to type 2 messages from 2 sources at same time unless deliberately trying to break something